### DISCUSSION TITLE
[LINUX / DEBIAN] How to compile 'TOOLS'?

### DISCUSSION THREAD

**Discussion by virgo77** [azerothcore/azerothcore-wotlk#12077] [General]
Hi there,
I'm trying to compile the TOOLS (maps extractors) under Linux Debian but it seems that TOOLS are not available yet.

As I'm trying this command: `cmake ../ -DCMAKE_INSTALL_PREFIX=$HOME/server.modded/ -DCMAKE_C_COMPILER=/usr/bin/clang -DCMAKE_CXX_COMPILER=/usr/bin/clang++ -DWITH_WARNINGS=1 -DTOOLS=1 -DSCRIPTS=static -DMODULES=static`
I get this strange information:
```
CMake Warning:
  Manually-specified variables were not used by the project:

    TOOLS
```

I searched on the wiki and did not find any solution, or the good command to launch. Is there a solution?

Thanks a lot :)

V.

**Comment by acidmanifesto** [comment]
Not entirely certain with linux. But on windows we have to set the following cmake manually for tools and the individual tools.
![unknown-406.png](https://user-images.githubusercontent.com/16887899/174415148-d5faf8a5-eb90-4be9-bbbe-3f9b09720dd3.png)

**Comment by virgo77** [comment]
ok thanks for your reply. I was on Windows and I leaved it for Debian because of less resources usage. ^^
Maybe I need to change the command line. Or is it inside the 'build' folder I need to change something? I'll search for it
Thanks a lot

**Comment by virgo77** [comment]
FOUND IT! :)
Thank you very much 'acidmanifsto'! You gave me the way to what I would like to do, it's great!

So, for devs, can you adapt your wiki please, with this little tuto?

> Before making any change, go to the 'build' folder, clean it, then run once this cmake command:
cmake ../ -DCMAKE_INSTALL_PREFIX=$HOME/folder.of.your.server/ -DCMAKE_C_COMPILER=/usr/bin/clang -DCMAKE_CXX_COMPILER=/usr/bin/clang++ -DWITH_WARNINGS=1 -DTOOLS=0 -DSCRIPTS=static -DMODULES=static

> It will create a fresh and clean cmake pre-compile content. Then do this, in order:


	1. In the 'build' folder, open the 'CMakeCache.txt' with Notepad++
	
	2. Search around lines 695/700 (at the end of the file), these lines:
		//STRINGS property for variable: TOOLS_BUILD
		TOOLS_BUILD-STRINGS:INTERNAL=none;all;db-only;maps-only
	
	3. Just know what these lines are (DON'T CHANGE ANYTHING HERE!):
		none= cmake won't create any of available TOOLS
		all= will create ALL TOOLS (dbimport / maps)
		db-only= will create DBIMPORT TOOL (which is the tool that will import your DB (maybe from TrinityCore to AzerothCore (I'm not sure about this one), or the AzerothCore DB.))
		maps-only= will create all MAPS Extractors (map_extractor / mmaps_generator / vmap4_assembler / vmap4_extractor)
	
	4. Now, pick the one you want to create
	
	5. Then, go around lines 375 / 380 and search for these lines:
		//Build list for tools
		TOOLS_BUILD:STRING=none
	
	6. Change the value to:
		TOOLS_BUILD:STRING=maps-only
	
	7. Save the file and quit
	
	8. In your CMAKE command line, change the '-DTOOLS=0' to '-DTOOLS=1', and launch it. In the pre-compile results, you will see these informations:
		* Tools build list (maps-only):
		|
		|  disabled
		|   +- dbimport
		|
		+- tools
		|   +- map_extractor
		|   +- mmaps_generator
		|   +- vmap4_assembler
		|   +- vmap4_extractor
		|
	
	9. Then run the 'make' and 'make install' commands. You will get the 4 extractors in the BIN folder of your server directory
	
	10. Just copy the extractors to the root of your WoW WotLK game folder
	
	11. Go to your game folder, then make sure that your 'Data' folder is wrote 'Data' and not 'data', even you will get an error:
		"No locale found"
	
	12. Then launch each command in this order (I didn't found the automatic .SH script. And the .BAT is not working):
		mkdir mmaps vmaps
		./map_extractor
		./vmap4_extractor
		./vmap4_assembler Buildings vmaps
		./mmaps_generator (be patient for this one. It's a bit long)
	
	13. At last, move these folders to the 'data' folder of your server/core: 
		Cameras
		dbc
		maps
		mmaps
		vmaps

	14. Remove the 'Buildings' folder:
		rm -Rf Buildings

	15. Now, go to your game server folder. Run it. You're done!


> That's all. Don't forget to do this again each time you need to rebuild the cmake files, if you always need the extractors

Have fun! :)

V.

**Comment by Winfidonarleyan** [comment]
```
cmake ../ -DCMAKE_INSTALL_PREFIX=/home/warhead -DAPPS_BUILD=none -DTOOLS_BUILD=all
```

@virgo77 just use `-DTOOLS_BUILD` build option if you use linux

**Reply by virgo77** [reply]
Hi Winfidonarleyan! Thanks for your reply. It was more easy ^^
I don't saw it on the wiki. Did I missed it? :-/

**Reply by Winfidonarleyan** [reply]
> I don't saw it on the wiki. Did I missed it? :-/

This is not yet in the wiki

**Reply by virgo77** [reply]
Hi Winfidonarleyan! Thanks again for your reply :)

Just another question: please, can you explain what 'DAPPS' are? Are these 'APPS' the dbimport option, or are these tools different?
Maybe I can just wait a wiki update too :)

Bye,

V.
